app/subscriber/src/features/my-folders/ConfigureFolder.tsx

```diff
@@ -63,22 +59,11 @@ export const ConfigureFolder: React.FC<IConfigureFolderProps> = () => {
   }, []);
 
   React.useEffect(() => {
-    if (currentFolder && !currentFolder.events.length) {
-      setCurrentFolder({
-        ...currentFolder,
-        events: [createSchedule(currentFolder.name, currentFolder.description)],
-      });
-      if (currentFolder.filterId) {
-        const selectedFilter = myFilters.find((f) => f.id === currentFolder.filterId);
-        setActiveFilter(selectedFilter);
-      }
-    }
-    // only run when currentFolder changes
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [currentFolder]);
+    setInit(true);
+  }, [id]);
 
   React.useEffect(() => {
-    if (init && ((!currentFolder && id) || currentFolder?.id !== Number(id))) {
+    if (init && myFolders.length && currentFolder?.id !== Number(id)) {
```

Comment:
We need to fetch the folder any time the url changes.  But we don't want to double fetch.